From fbf250441db77c3eaa0f02bf4fa077fabea89c18 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 12 May 2020 23:37:47 +0300
Subject: [PATCH] [SW-0] preboot: clean up uboot left overs

The slim Preboot variant is going to load various images and not only
U-Boot. Thus rename all "uboot" instances into a more generic names.

Change-Id: Iefb4c93058f13b4eb26de5a16503085f47d76603
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/cmd_module.h b/common/cmd_module.h
index 3a0e26d..680f10e 100644
--- a/common/cmd_module.h
+++ b/common/cmd_module.h
@@ -79,7 +79,7 @@
 void module_lkd_disable(void);
 void module_lkd_run(void);
 void module_lkd_run_noret(void);
-void uboot_start(u64 loadaddr);
+void image_start(u64 loadaddr);
 
 /* UART */
 enum cmd_check_res module_uart_check(void);
diff --git a/common/fit.c b/common/fit.c
index 8b9ffb3..1b468d6 100644
--- a/common/fit.c
+++ b/common/fit.c
@@ -78,7 +78,7 @@
 	return 1;
 }
 
-int fit_load_uboot(const void *fit, u64 *dest)
+int fit_load_image(const void *fit, u64 *dest)
 {
 	int node, len;
 	const u8 *buf;
@@ -87,13 +87,13 @@
 
 	node = fit_loadables_node_get(fit, 0);
 	if (node < 0) {
-		preboot_puts("Wrong FIT format: no uboot node\n");
+		preboot_puts("Wrong FIT format: no loadable image node\n");
 		return 1;
 	}
 
 	loadaddr = fdt_getprop(fit, node, FIT_LOADADDR_PROP, &len);
 	if (!loadaddr || len != sizeof(u64)) {
-		preboot_puts("Wrong FIT format: no uboot loadaddr\n");
+		preboot_puts("Wrong FIT format: no image loadaddr\n");
 		return 1;
 	}
 
@@ -102,7 +102,7 @@
 
 	buf = fdt_getprop(fit, node, FIT_DATA_PROP, &len);
 	if (!buf) {
-		preboot_puts("Wrong FIT format: uboot data empty\n");
+		preboot_puts("Wrong FIT format: image data empty\n");
 		return 1;
 	}
 
diff --git a/common/fit.h b/common/fit.h
index f02271b..ed9919a 100644
--- a/common/fit.h
+++ b/common/fit.h
@@ -8,7 +8,7 @@
 #ifndef _HL_FIT_H_
 #define _HL_FIT_H_
 
-int fit_load_uboot(const void *fit, u64 *dest);
+int fit_load_image(const void *fit, u64 *dest);
 int fit_check_format(const void *fit);
 void fit_copy_image(u64 dst, const void *buf, int len);
 
diff --git a/common/module_lkd.c b/common/module_lkd.c
index a0bd721..f067ad0 100644
--- a/common/module_lkd.c
+++ b/common/module_lkd.c
@@ -69,11 +69,11 @@
 void module_lkd_run(void)
 {
 	int err;
-	void *fit = (void *)UBOOT_FIT_BASE;
+	void *fit = (void *)FIT_IMG_BASE;
 
 	err = !fit_check_format(fit);
 	if (!err)
-		err = fit_load_uboot(fit, &sts.loadaddr);
+		err = fit_load_image(fit, &sts.loadaddr);
 
 	if (err) {/* Report the FIT is bad */
 		cpu_boot_err_clrset(CPU_BOOT_ERR0_FIT_CORRUPTED, true);
@@ -94,7 +94,7 @@
 		wfe();
 		break;
 	case KMD_MSG_FIT_RDY:
-		uboot_start(sts.loadaddr);
+		image_start(sts.loadaddr);
 		/* no break */
 	default:
 		break;
diff --git a/gaudi/common.c b/gaudi/common.c
index 8afd619..7ef6d12 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -216,11 +216,11 @@
 	preboot_puts("5\n");
 }
 
-void uboot_start(u64 loadaddr)
+void image_start(u64 loadaddr)
 {
-	void (*uboot_entry)(void) = (void *)loadaddr;
+	void (*image_entry)(void) = (void *)loadaddr;
 
-	uboot_entry();
+	image_entry();
 }
 
 void preboot_start(u64 loadaddr)
diff --git a/goya/common.c b/goya/common.c
index 4a27239..22f7bfd 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -168,11 +168,11 @@
 
 void hl_plat_init(void) {}
 
-void uboot_start(u64 loadaddr)
+void image_start(u64 loadaddr)
 {
-	void (*uboot_entry)(void) = (void *)loadaddr;
+	void (*image_entry)(void) = (void *)loadaddr;
 
-	uboot_entry();
+	image_entry();
 }
 
 void preboot_start(u64 loadaddr)
diff --git a/goya2/common.c b/goya2/common.c
index 050cd08..6ed079b 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -225,11 +225,11 @@
 	*regs = mmPSOC_QSPI_CTRLR0;
 }
 
-void uboot_start(u64 loadaddr)
+void image_start(u64 loadaddr)
 {
-	void (*uboot_entry)(void) = (void *)loadaddr;
+	void (*image_entry)(void) = (void *)loadaddr;
 
-	uboot_entry();
+	image_entry();
 }
 
 void preboot_start(u64 loadaddr)
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index eeb8dc6..545cc46 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -287,7 +287,7 @@
 	memcpy((void *)arc_dst, buf, len);
 }
 
-void uboot_start(u64 loadaddr)
+void image_start(u64 loadaddr)
 {
 	u32 lsb = (u32)loadaddr, msb = (u32)(loadaddr >> 32);
 
diff --git a/include/preboot.h b/include/preboot.h
index 72e8b05..e05b4f4 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -15,7 +15,7 @@
 #define HL_SPI_BASE				0x1A000000
 #define HL_CFG_BASE				0xFC000000
 #define SDRAM_BASE				0x10000000
-#define UBOOT_FIT_BASE				0x10200000
+#define FIT_IMG_BASE				0x10200000
 
 #define wfe()	__asm__ __volatile__("sleep 0x11\n\t" : : )
 #endif
@@ -24,7 +24,7 @@
 #define HL_SPI_BASE				0x7FF8000000ull
 #define HL_CFG_BASE				0x7FFC000000ull
 #define SDRAM_BASE				0x7FF0000000ull
-#define UBOOT_FIT_BASE				0x7FF0200000ull
+#define FIT_IMG_BASE				0x7FF0200000ull
 
 #define wfe()	__asm__ __volatile__ ("wfe\n\t" : : )
 #endif
